primitives/fixed.fil

```diff
@@ -0,0 +1,12 @@
+/**
+ * Defines combinational fixed point math operations
+ */
+
+extern "fixed.sv" {
+  comp FXPMult[W, ?D=W/2]<'G: 'L-('G), ?'L: 1='G+1>(
+    left: ['G, 'L] W,
+    right: ['G, 'L] W,
+  ) -> (
+    out: ['G, 'L] W
+  ) where 'L > 'G, D < W, W > 0;
```

Comment:
Why do we need an extern for this? Fixed point multiplication can be implemented in terms of standard multiplication right. The Calyx primitives library implements `std_mult_pipe` (integer mult) in `std_fp_mult_pipe` (fixed point mult): https://github.com/cucapra/calyx/blob/master/primitives/binary_operators.sv#L366

Reply:
Wouldn't we still have to do the right shift at the end to discard the bottom `d` bits? Not sure how verilog can do fixed point multiplication.

Reply:
It doesn't have to be a primitive I guess, I could implement it as a component instead.

Reply:
Yeah, take a look at how Calyx does this. Fewer things as primitives is better because then Filament can make strong guarantees about the program.